### PR TITLE
[DQM] [CLANG] Fixed unused-but-set-variable warnings

### PR DESCRIPTION
**Pull request by aandvalenzuela**
Follow-up PR for https://github.com/cms-sw/cmssw/pull/40400.
This PR fixes `unused-but-set-variable` warnings which we get with LLVM14 in CLANG IBS

### CODE REVIEW

**Comment by tvami** [comment]
hi @aandvalenzuela you commented out some of the unused code, but I think it should just be removed fully

**Comment by aandvalenzuela** [comment]
Hi @tvami,
Thanks, I can remove it. I didn't remove these lines since there is some already commented out code that also uses these variables. So in case someone uncomments these lines, there won't be trace of this variable. Should I also remove the already commented code?

**Comment by emanueleusai** [comment]
I would go ahead and delete the already commented code as well.

**Comment by aandvalenzuela** [comment]
Done! @emanueleusai, @tvami

File: DQM/BeamMonitor/plugins/FakeBeamMonitor.cc [hunk 1]
```diff
@@ -891,26 +889,6 @@ void FakeBeamMonitor::FitAndFill(const LuminosityBlock& lumiSeg, int& lastlumi,
...
-  //  } else {
-  //    for (size_t i = 0; i < 7; i++) {
-  //      if (i > 0) {
-  //        DipPVInfo_.push_back(0.);
-  //      } else {
-  //        DipPVInfo_.push_back((float)countEvtLastNLS_);
-  //      }
-  //    }
-  //  }
-  //  theBeamFitter->SetPVInfo(DipPVInfo_);
```
Comment (mmusich) [inline DQM/BeamMonitor/plugins/FakeBeamMonitor.cc:913]: @francescobrivio, please confirm it's fine to remove this commented section.

Comment (francescobrivio) [reply DQM/BeamMonitor/plugins/FakeBeamMonitor.cc:913]: Thanks Marco for reminding me!
Yes it's fine to remove these commented lines from the BeamSpot point of view!

Comment (aandvalenzuela) [reply DQM/BeamMonitor/plugins/FakeBeamMonitor.cc:913]: Thanks @mmusich, @francescobrivio

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
+db
 * db part of the removal confirmed by the experts https://github.com/cms-sw/cmssw/pull/40513#pullrequestreview-1249912994

**Comment by cecilecaillol** [comment]
+l1

**Comment by emanueleusai** [comment]
+1

**Comment by perrotta** [comment]
@emanueleusai there quite several differences in the L1TEMU/L1TStage2CaloLayer2 DQM plots for HI 2018 workflow 140.56, which look correlated: could you please have a look?

File: DQMOffline/CalibTracker/plugins/SiStripPopConBadComponentsDQM.cc [hunk 2]
```diff
@@ -103,7 +101,6 @@ void SiStripPopConBadComponentsHandlerFromDQM::dqmEndJob(DQMStore::IBooker&, DQM
 
     std::vector<std::string> module_folders;
     getModuleFolderList(getter, dname, module_folders);
-    int nDets = module_folders.size();
 
     int nDetsWithError = 0;
```
Comment (perrotta) [inline DQMOffline/CalibTracker/plugins/SiStripPopConBadComponentsDQM.cc:105]: `nDetsWithError` also becomes useless now

File: DQM/PixelLumi/plugins/PixelLumiDQM.cc [hunk 3]
```diff
@@ -430,7 +429,6 @@ void PixelLumiDQM::endLuminosityBlock(edm::LuminosityBlock const &lumiBlock, edm
                     (*it).second.dnumFP.at(1) + (*it).second.dnumFM.at(0) + (*it).second.dnumFM.at(1);
     if (useInnerBarrelLayer)
       etotal = (*it).second.dnumB.at(0);
```
Comment (perrotta) [inline DQM/PixelLumi/plugins/PixelLumiDQM.cc:431]:
```suggestion
    double etotal = useInnerBarrelLayer 
                              ? (*it).second.dnumB.at(0)
                              : (*it).second.dnumB.at(1) + (*it).second.dnumB.at(2) + (*it).second.dnumFP.at(0) +
                                (*it).second.dnumFP.at(1) + (*it).second.dnumFM.at(0) + (*it).second.dnumFM.at(1);
```
can avoid some useless computation if `useInnerBarrelLayer`

**Review by perrotta — COMMENTED** [review]
I did not find the possible origin of the differences in the DQM plots, but I spotted a couple of possible optimizations instead

**Comment by aandvalenzuela** [comment]
please test

**Comment by aandvalenzuela** [comment]
please test

**Comment by tvami** [comment]
+db
 * resign

**Comment by cecilecaillol** [comment]
+l1

**Comment by emanueleusai** [comment]
> @emanueleusai there quite several differences in the L1TEMU/L1TStage2CaloLayer2 DQM plots for HI 2018 workflow 140.56, which look correlated: could you please have a look?

@perrotta I do not see the differences in that workflow in the latest comparison. Could you please link to where you see the differences?

**Comment by perrotta** [comment]
> @perrotta I do not see the differences in that workflow in the latest comparison. Could you please link to where you see the differences?

Last comparisons look clean, indeed. And also the past ones still available in this thread, as far as I can see: I suspect that I was looking at a wrong page when I made that comment: sorry for the noise

**Comment by emanueleusai** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**